jdaviz/configs/default/plugins/plot_options/plot_options.py

```diff
@@ -753,135 +767,82 @@ def _update_stretch_histogram_colorbar(self, msg={}):
-                    mark.x = []
-                    mark.y = []
-            return
-
-        for layer in self.layer.selected_obj:
-            if isinstance(layer.layer, GroupedSubset):
-                # don't update histogram for subsets:
-                continue
-
-            # clear old mark, if it exists:
-            mark_label = f'{mark_label_prefix}{layer.label}'
-            mark_exists = mark_label in self.stretch_histogram.marks
-
-            knot_label = f"{knots_label_prefix}{layer.label}"
-            # create the new/updated mark following the colormapping
-            # procedure in glue's CompositeArray:
-            interval = ManualInterval(self.stretch_vmin.value, self.stretch_vmax.value)
-            contrast_bias = ContrastBiasStretch(layer.state.contrast, layer.state.bias)
-            stretch = stretches.members[layer.state.stretch]
-            layer_cmap = layer.state.cmap
-
-            if isinstance(stretch, SplineStretch):
-                knot_x = (self.stretch_vmin_value +
-                          stretch.x * (self.stretch_vmax_value - self.stretch_vmin_value))
-                knot_y = stretch.y
+        # create the new/updated stretch curve following the colormapping
+        # procedure in glue's CompositeArray:
+        interval = ManualInterval(self.stretch_vmin_value, self.stretch_vmax_value)
+        contrast_bias = ContrastBiasStretch(self.image_contrast_value, self.image_bias_value)
+        stretch = stretches.members[self.stretch_function_value]
+        layer_cmap = layer.state.cmap
+
+        # show the colorbar
+        color_mode = self.image_color_mode_value
+
+        # NOTE: Index 0 in marks is assumed to be the bin centers.
+        x = self.stretch_histogram.figure.marks[0].x
+        y = np.ones_like(x)
+
+        # Copied from the __call__ internals of glue/viewers/image/composite_array.py
+        data = interval(x)
+        data = contrast_bias(data, out=data)
+        data = stretch(data, out=data)
+
+        if color_mode == 'Colormaps':
+            cmap = colormaps[self.image_colormap.text]
+            if hasattr(cmap, "get_bad"):
+                bad_color = cmap.get_bad().tolist()[:3]
+                layer_cmap = cmap.with_extremes(bad=bad_color + [self.image_opacity_value])
             else:
-                knot_x, knot_y = [], []
+                layer_cmap = cmap
+
+            # Compute colormapped image
+            plane = layer_cmap(data)
+
+        else:  # Monochromatic
+            # Get color
+            color = COLOR_CONVERTER.to_rgba_array(self.image_color_value)[0]
+            plane = data[:, np.newaxis] * color
+            plane[:, 3] = 1
+
+        plane = np.clip(plane, 0, 1, out=plane)
+        ipycolors = [matplotlib.colors.rgb2hex(p, keep_alpha=False) for p in plane]
+
+        colorbar_mark = self.stretch_histogram.marks['colorbar']
+        colorbar_mark.x = x
+        colorbar_mark.y = y
+        # TODO: fix all white on load...

```

Comment:
Is this TODO a future PR or something you want to fix as part of this PR?

Reply:
woops, I broke and fixed that in this PR, I'll remove the comment.